### PR TITLE
feat: add token detail field to response schema of Lock Events API

### PR DESCRIPTION
**Pull request by purplesmoke05**
Close: #1355

### CODE REVIEW

File: app/api/routers/position.py [hunk 1]
```diff
@@ -125,7 +134,10 @@ def __call__(
         sort_item = request_query.sort_item
         sort_order = request_query.sort_order  # default: asc
 
-        query = session.query(IDXLockedPosition)
+        query = session.query(IDXLockedPosition, self.idx_token_model).join(
+            self.idx_token_model,
+            IDXLockedPosition.token_address == self.idx_token_model.token_address,
+        )
```
Comment (purplesmoke05) [inline app/api/routers/position.py:140]: - Added INNER JOIN clause to LockedPosition query.

File: app/api/routers/position.py [hunk 2]
```diff
@@ -157,11 +169,14 @@ def __call__(
...
-        _locked_list: list[IDXLockedPosition] = query.all()
+        _locked_list: list[tuple[IDXLockedPosition, IDXTokenInstance]] = query.all()
         locked_list = []
 
         for _locked in _locked_list:
-            locked_list.append(_locked.json())
+            _locked_data = _locked[0].json()
+            if request_query.include_token_details is True:
+                _locked_data["token"] = self.token_model.from_model(_locked[1]).__dict__
+            locked_list.append(_locked_data)
```
Comment (purplesmoke05) [inline app/api/routers/position.py:179]: - If **include_token_details=true** is queried, token_detail attribute is given.

File: app/api/routers/position.py [hunk 3]
```diff
@@ -219,6 +243,11 @@ def __call__(
             IDXLock.value.label("value"),
             IDXLock.data.label("data"),
             IDXLock.block_timestamp.label("block_timestamp"),
+            self.idx_token_model,
+        ).join(
+            self.idx_token_model,
+            IDXLock.token_address.label("token_address")
+            == self.idx_token_model.token_address,
```
Comment (purplesmoke05) [inline app/api/routers/position.py:250]: - Added INNER JOIN clause to LockEvent query.

File: app/api/routers/position.py [hunk 4]
```diff
@@ -285,21 +319,24 @@ def __call__(
...
+                "value": lock_event[6],
+                "data": lock_event[7],
+                "block_timestamp": lock_event[8]
+                .replace(tzinfo=UTC)
+                .astimezone(local_tz),
+            }
+            if request_query.include_token_details is True:
+                event_data["token"] = self.token_model.from_model(
+                    lock_event[9]
+                ).__dict__
```
Comment (purplesmoke05) [inline app/api/routers/position.py:338]: - If **include_token_details=true** is queried, token_detail attribute is given.

File: app/model/schema/position.py [hunk 5]
```diff
@@ -166,6 +174,9 @@ class ListAllLockedPositionQuery:
     token_address_list: list[StrictStr] = Query(
         default=[], description="list of token address (**this affects total number**)"
     )
+    include_token_details: Optional[bool] = Query(
+        default=False, description="include token details"
+    )
```
Comment (purplesmoke05) [inline app/model/schema/position.py:179]: - Added **include_token_details** query param.
  - Default value is `False`

File: app/model/schema/position.py [hunk 6]
```diff
@@ -199,6 +210,9 @@ class ListAllLockEventQuery:
     sort_order: SortOrder = Query(
         default=SortOrder.DESC, description="sort order(0: ASC, 1: DESC)"
     )
+    include_token_details: Optional[bool] = Query(
+        default=False, description="include token details"
+    )
```
Comment (purplesmoke05) [inline app/model/schema/position.py:215]: - Added **include_token_details** query param.
  - Default value is `False`

File: tests/app/position_PositionShare_Lock_Event_GET_test.py [hunk 7]
```diff
@@ -153,7 +237,15 @@ def setup_data(self, session: Session, token_address: str, base_time: datetime):
...
     # List all Events
-    def test_normal_1(self, client: TestClient, session: Session):
+    @pytest.mark.parametrize(
+        "get_params",
+        [
+            {"include_token_details": True},
+            {"include_token_details": False},
+            {},
+        ],
+    )
```
Comment (purplesmoke05) [inline tests/app/position_PositionShare_Lock_Event_GET_test.py:247]: Made test cases parametrize to ensure both of response data(token_detail is include and not) is valid.

**Review by YoshihitoAso — APPROVED** [review]

**Pull request merged**